public/app/features/dashboard-scene/panel-edit/PanelDataPane/PanelDataQueriesTab.tsx

```diff
@@ -131,10 +189,34 @@ function PanelDataQueriesTabRendered({ model }: SceneComponentProps<PanelDataQue
         data={data}
         queries={model.getQueries()}
         dsSettings={dsSettings}
-        onAddQuery={() => {}}
+        onAddQuery={model.onAddQuery}
         onQueriesChange={model.onQueriesChange}
         onRunQueries={model.onRunQueries}
       />
+
+      <HorizontalGroup spacing="md" align="flex-start">
+        {showAddButton && (
+          <Button
+            icon="plus"
+            onClick={model.addQueryClick}
+            variant="secondary"
+            data-testid={selectors.components.QueryTab.addQuery}
+          >
+            Add query
+          </Button>
+        )}
+        {config.expressionsEnabled && model.isExpressionsSupported(dsSettings) && (
+          <Button
+            icon="plus"
+            onClick={model.onAddExpressionClick}
+            variant="secondary"
+            data-testid="query-tab-add-expression"
+          >
+            <span>Expression&nbsp;</span>
+          </Button>
+        )}
+        {model.renderExtraActions()}
```

Comment:
i've tried running all possible scenarios in gdev datasource but didn't see any extra actions displayed. not sure how to test this

Reply:
Dod you learn how to get those up?

Reply:
Yes, it's an enterprise feature called record queries which once configured at a `Recorded Query` button. Here's a little demo:


https://github.com/grafana/grafana/assets/58232930/1ba613a0-1b36-4416-a5db-0015d27c125e

I didn't find any additional external actions